packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx

```diff
@@ -105,38 +328,393 @@ export class SankeyChartBase extends React.Component<
+                  textAnchor="start"
+                  fontWeight="bold"
+                  fill={
+                    !(
+                      !this.state.selectedState ||
+                      (this.state.selectedState &&
+                        this.state.selectedNodes.indexOf(singleNode) !== -1 &&
+                        this.state.selectedNode) ||
+                      (this.state.selectedState && !this.state.selectedNode)
+                    )
+                      ? '#323130'
+                      : '#FFFFFF'
+                  }
+                  fontSize={14}
+                >
+                  {singleNode.actualValue}
+                </text>
+              </g>
+            )}
           </g>
         );
         nodes.push(node);
       });
       return nodes;
     }
   }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private _onLeave(singleNode: any) {
+    if (this.state.selectedState) {
+      this.setState({
+        selectedState: false,
+        selectedNodes: [],
+        selectedLinks: [],
+        selectedNode: null,
+      });
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private _onHover(singleNode: any, mouseEvent: React.MouseEvent<SVGElement>) {
+    mouseEvent.persist();
+    if (!this.state.selectedState) {
+      const selectedLinks = this._getSelectedLinks(singleNode);
+      const selectedNodes = this._getSelectedNodes(selectedLinks);
+      selectedNodes.push(singleNode);
+      this.setState({
+        selectedState: true,
+        selectedNodes,
+        selectedLinks,
+        selectedNode: singleNode,
+        nodeElement: mouseEvent,
+      });
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private _onStreamHover(singleLink: any, mouseEvent: React.MouseEvent<SVGElement>) {
+    mouseEvent.persist();
+    if (!this.state.selectedState) {
+      const { selectedLinks, selectedNodes } = this._getSelectedLinksforStreamHover(singleLink);
+      this.setState({
+        selectedState: true,
+        selectedNodes,
+        selectedLinks,
+        linkElement: mouseEvent,
+        selectedLink: singleLink,
+      });
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private _onStreamLeave(singleLink: any) {
+    if (this.state.selectedState) {
+      this.setState({
+        selectedState: false,
+        selectedNodes: [],
+        selectedLinks: [],
+      });
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private _fillNodeColors(singleNode: any): string {
+    if (!this.state.selectedState) {
+      return singleNode.color;
+    } else if (
+      this.state.selectedState &&
+      this.state.selectedNodes.indexOf(singleNode) !== -1 &&
+      this.state.selectedNode
+    ) {
+      return this.state.selectedNode.color;
+    } else if (this.state.selectedState && !this.state.selectedNode) {
```

Comment:
What does selected state mean

Reply:
Selected state means that some selection(mouse hover) is done in the chart either node or link selection.